test/CMakeLists.txt

```diff
@@ -1,100 +1,46 @@
 # SPDX-License-Identifier: Apache-2.0
-
-#####################################################
-#################### GETEST #######################
-#####################################################
-enable_testing()
-if(KOMPUTE_OPT_REPO_SUBMODULE_BUILD)
-    add_subdirectory(${PROJECT_SOURCE_DIR}/external/googletest EXCLUDE_FROM_ALL
-        ${CMAKE_CURRENT_BINARY_DIR}/kompute_googletest)
-else()
-    find_package(GTest CONFIG REQUIRED)
-endif()
-
-file(GLOB test_kompute_CPP
-    "${CMAKE_CURRENT_SOURCE_DIR}/*.cpp"
-)
-
-add_executable(test_kompute ${test_kompute_CPP})
-
-target_include_directories(
-    test_kompute PUBLIC
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/single_include>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/compiled_shaders_include>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/utils>
-)
-
-if(KOMPUTE_OPT_REPO_SUBMODULE_BUILD)
-    target_include_directories(
-        test_kompute PRIVATE
-        ${gtest_SOURCE_DIR}/include)
-
-    target_link_libraries(test_kompute
-        gtest_main)
-else()
-    target_link_libraries(test_kompute
-        GTest::gtest)
-endif()
-
-target_link_libraries(test_kompute kompute)
-
-add_test(NAME test_kompute COMMAND test_kompute)
-
-
-#####################################################
-#################### CODECOV  #######################
-#####################################################
-
-if (KOMPUTE_OPT_CODE_COVERAGE)
-    if(NOT UNIX)
-        message(
-            FATAL_ERROR
-            "KOMPUTE_OPT_CODE_COVERAGE can only be enabled in unix based systems due to limitation on gcov")
+# ######################
+cmake_minimum_required(VERSION 3.14)
+
+# ####################################################
+# Shaders
+# ####################################################
+add_subdirectory(shaders)
+
+# ####################################################
+# Tests
+# ####################################################
+macro(add_kompute_test _TEST_NAME)
+    add_executable("${_TEST_NAME}_tests" "Test${_TEST_NAME}.cpp"
```

Comment:
It seems there may be an issue with having this as separate executables as opposed  to all as a single one, namely it seems each executable would have to be called independently, as opposed to being able to call the full test suite and then filter by test names - how are you currently running all tests vs a subset of tests across multiple files?